lib/tasks/scripts/add_permission_to_access_level.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Creates the specified permission for users who have the specified access_level
+class AddPermissionToAccessLevel
+
+  attr_reader :permission_name, :access_level_name, :access_level, :permission
+
+  def initialize(permission_name, access_level_name)
+    @permission_name = permission_name
+    @access_level_name = access_level_name
+    @permission = Permissions::ALL_PERMISSIONS[permission_name]
+    @access_level = Permissions::ACCESS_LEVELS.select { |level| level[:name] == access_level_name }.first
+  end
+
+  def valid?
+    permission && valid_permission_for_access_level?
+  end
+
+  def create
+    users.each do |user|
+      permission_resources(user).each do |resource|
+        UserPermission
+          .where(user: user,
+                 permission_slug: permission[:slug],
+                 resource_type: resource[:resource_type],
+                 resource_id: resource[:resource_id])
+          .first_or_create
+      end
+    end
+  end
+
+  private
+
+  def users
+    users = User.includes(:user_permissions).where.not(user_permissions: { id: nil })
+    users.select{ |user| user_is_a_match?(user) }
+  end
+
+  def permission_resources(user)
+    return [{ resource_type: nil, resource_id: nil }] if permission[:resource_priority] == [:global]
+
+    existing_resource_types = user.user_permissions.map(&:resource_type).uniq
+
+    case
+    when permission[:resource_priority].include?(:facility_group) && existing_resource_types.include?('FacilityGroup')
+      user.user_permissions.where(resource_type: 'FacilityGroup').map do |resource|
+        { resource_type: resource.resource_type, resource_id: resource.resource_id }
+      end.uniq
+    when permission[:resource_priority].include?(:organization) && existing_resource_types.include?('Organization')
+      user.user_permissions.where(resource_type: 'Organization').map do |resource|
+        { resource_type: resource.resource_type, resource_id: resource.resource_id }
+      end.uniq
+    when permission[:resource_priority].include?(:global) && existing_resource_types.include?(nil)
+      [{ resource_type: nil, resource_id: nil }]
+    end
```

Comment:
Hmm, I get that, but 
- Its possible for a user to have say `view_overdue_list` on multiple facility groups, I've seen this in our data. The user would still have the same `access_level/role`.
- Some existing roles create permissions with different scopes. Eg: The `supervisor` role (CVHOs) have mostly `facility_group` permissions, but one `organization` permission. In this case its so that they can `manage_admins` and invite other people to the dashboard.

Reply:
After our Slack discussion, this seems to make great sense 👌 Nice.